src/connector.js

```diff
@@ -141,417 +92,654 @@ let pageTemplateHTML = document.getElementById( 'sr-pager-page-' )?.innerHTML;
+				aqString += ' @filetype==(ps)';
+			}
+			else if ( filetype === "application/msword" ) {
+				aqString += ' @filetype==(doc,docx)';
+			}
+			else if ( filetype === "application/vnd.ms-excel" ) {
+				aqString += ' @filetype==(xls,xlsx)';
+			}
+			else if ( filetype === "application/vnd.ms-powerpoint" ) {
+				aqString += ' @filetype==(ppt,pptx)';
+			}
+			else if ( filetype === "application/rtf" ) {
+				aqString += ' @filetype==(rtf)';
+			}
 		}
+
+		if ( urlParams.site ) {
+			let site = urlParams.site.toLowerCase().replace( '*', '' );
+			aqString += ' @canadagazettesite==' + site;
+		}
+
+		if ( aqString ) {
+			const action = loadAdvancedSearchQueryActions( headlessEngine ).updateAdvancedSearchQueries( { 
+				aq: aqString,
+			} );
+			headlessEngine.dispatch( action ); 
+		}
+
+		searchBoxController.updateText( qString );
+		searchBoxController.submit();
 	}
-	if ( urlParams.dmn ) {
-		aqString += ' @hostname="' + urlParams.dmn + '"';
-	}
-	
-	if ( urlParams.sort ) {
-		const sortAction = loadSortCriteriaActions( headlessEngine ).updateSortCriterion( {
-			criterion: "SortByDate",
-		} );
-		headlessEngine.dispatch( sortAction );
+
+	if ( hashParams.q && searchBoxElement ) {
+		searchBoxElement.value = hashParams.q;
 	}
-	
-	if ( aqString ) {
-		const action = loadAdvancedSearchQueryActions( headlessEngine ).updateAdvancedSearchQueries( { 	
-			aq: aqString,
-		} );
-		headlessEngine.dispatch( action ); 
+	else if ( urlParams.q && searchBoxElement ) {
+		searchBoxElement.value = urlParams.q;
 	}
-	
-	searchBoxController.updateText( qString );
-	searchBoxController.submit();
-}
 
-if ( hashParams.q && searchBoxElement ) {
-	searchBoxElement.value = hashParams.q;
-}
-else if ( urlParams.q && searchBoxElement ) {
-	searchBoxElement.value = urlParams.q;
-}
+	// Get the query portion of the URL
+	const fragment = () => {
+		const hash = window.location.hash.slice( 1 );
+		if (!statusController.state.firstSearchExecuted && !hashParams.q ) {
+			return window.location.search.slice( 1 ).replaceAll( '+', ' ' ); // use query string if hash is empty
+		}
 
-// Get the query portion of the URL
-const fragment = () => {
-	const hash = window.location.hash.slice( 1 );
-	if (!statusController.state.firstSearchExecuted && !hashParams.q ) {					
-		return window.location.search.slice( 1 ); // use query string if hash is empty
-	}
-	
-	return hash;
-};
+		return hash;
+	};
 
-const urlManager = buildUrlManager( headlessEngine, {
-	initialState: {
-		fragment: fragment(),
-	},
-} );
+	urlManager = buildUrlManager( headlessEngine, {
+		initialState: {
+			fragment: fragment(),
+		},
+	} );
 
-// Unsubscribe to controllers
-const unsubscribeManager = urlManager.subscribe( () => {
-	if ( !params.enableHistoryPush || window.location.origin.startsWith( 'file://' ) ) {
-		return;
-	}
+	// Unsubscribe to controllers
+	unsubscribeManager = urlManager.subscribe( () => {
+		if ( !params.enableHistoryPush || window.location.origin.startsWith( 'file://' ) ) {
```

Comment:
:scream:

Really, local file system????

There is so many way to store information locally, local storage, cache,......